### ISSUE TITLE
admin story: versions management

### ISSUE THREAD

**Issue by jackfrancis** [deis/workflow-manager#20]
As a deis cluster admin I need to be able to easily determine if the cluster components are up-to-date so that I can keep the cluster current with the latest stable versions.

**Comment by jackfrancis** [comment]
This issue should track a simple HTTP interface (e.g., a route handler) that exposes the complete set of installed deis components in a deis cluster, and whether or not those components have a newer version available to download/install. I.e., an implementation of `types.ComponentVersion` with a populated `UpdateAvailable` property.

**Comment by arschles** [comment]
note that we may want to add commands in workflow-cli to access this API. may need an issue in that repo to accompany this

**Comment by bacongobbler** [comment]
Yeah I'd like to see something like this implemented, but I wonder what our UX story is for this. Do we need to create a separate admin client so the administrator can access these APIs, or would the admin have to SSH into the cluster to consume this API? I don't think we'd want to expose workflow-manager out to the public unless it was backed by auth, for example.

**Comment by arschles** [comment]
@slack mind commenting here about the questions that @bacongobbler brought up above? Added context would be helpful here

**Comment by bacongobbler** [comment]
One way to tackle this would be to add an API endpoint to the controller that proxies the request over to workflow-manager and checks that the user making the request is an administrator, but at that point we might also want to consider breaking out the authentication module into its own in-cluster SSO.

**Comment by slack** [comment]
Since we don't yet have a workflow admin tool, the original thought was to use `kubectl proxy` to access workflow-manager on their k8s cluster. That follows the pattern for k8s dashboard, etc.

After authentication issues are sorted out, we could use deis-router to expose an interface. But that is well down the road, for now.

Back to the original design doc, the user stories here are:

**As a Workflow Manager User I should:**
* be able to view a simple web UI that I can reach through the kubernetes proxy
    * I should not have to authenticate to this UI
* be able to view a list of services, replication controllers and pods that are installed or running in the Deis namespace
    * so that I can quickly, at a glance, know if all the components are installed
* be able to see the versions of installed workflow components, and be notified if they are out of date
     * annotations, container images, (whatever makes sense here)
* see a large banner if deis workflow components are out of date
* poll a hosted releases API off-cluster for release information at least every 12 hours

So a data API and a simple web page that cluster admins can view to see installed components, versions and know if things are out of date.

At this point, I'd even be happy with workflow-manager sending version information to stdout so you could get at it with `kd get logs workflow-manager`.
